geometric_kernels/spaces/product.py

```diff
@@ -0,0 +1,195 @@
+"""
+Implement product spaces
+"""
+
+from typing import List
+
+import lab as B
+import numpy as np
+
+from .base import DiscreteSpectrumSpace
+
+from geometric_kernels.eigenfunctions import Eigenfunctions
+
+
+class ProductEigenfunctions(Eigenfunctions):
+    def __init__(
+        self,
+        dimensions: List[int],
+        eigenindicies: B.Numeric,
+        *eigenfunctions: Eigenfunctions,
```

Comment:
Why can this not be `List[Eigenfunctions]`? This would better match the `List[int]` of corresponding dimensions.